include/ozo/detail/timeout_handler.h

```diff
@@ -5,21 +5,24 @@
 
 namespace ozo::detail {
 
-template <typename Socket, typename Allocator>
+template <typename Connection, typename Allocator>
 struct cancel_socket {
-    Socket& socket_;
+    Connection conn_;
     Allocator allocator_;
 
-    cancel_socket(Socket& socket, const Allocator& allocator)
-    : socket_(socket), allocator_(allocator) {}
+    cancel_socket(Connection&& conn, const Allocator& allocator)
+    : conn_(conn), allocator_(allocator) {}
```

Comment:
std::move(conn)?

Reply:
**Done**

Thanks, but `std::forward` is really needed.